.github/workflows/python-test.yml

```diff
@@ -0,0 +1,25 @@
+name: Tests
+
+on:
+  pull_request:
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.9
```

Comment:
(optional) This could be also updated to latest stable if you are doing significant refactoring.

Reply:
@dobairoland Do you mean update the whole project for latest Python (3.12)? 

(Python 3.9 is default for `node-bullseye` Docker base image so this is why all is set with Py3.9 here.)

Reply:
> Do you mean update the whole project for latest Python (3.12)?

Yes, or whatever newer version is convenient for you. bullseye is oldstable now. Bookworm could be used where (I think) Python 3.11 is included.

Reply:
@dobairoland  Updated, thanks for the tip. (Tested with Bookworm + Py3.11 [here](https://github.com/espressif/test-project-bot/actions/runs/7872708175/job/21478490767))